### PR TITLE
Replace Restify with Express

### LINKED ISSUE #5: Switch JWT library to use Asynchronous verify

### PR DESCRIPTION
**Pull request by nwinch**
This branch resolves #5, #64 and #73.

Restify is gone.  Long live Express!

**A few things**:
- Middleware folder has been removed.  There were two - CORS and JWT; CORS is now part of index, and JWT is now using `express-jwt` instead the existing custom implementation.  Test files have been repurposed appropriately.
- CORS generator has been removed as it's now part of `/base`.
- Simple [error middleware](https://github.com/Prismatik/redbeard/compare/express?expand=1#diff-90ab418476e49872e9ad590c6b64aec6R25) has been added, though this could probably be something more robust like [`kiss-my-error`](https://github.com/Prismatik/kiss-my-error).

### CODE REVIEW

File: base/index.mustache [hunk 1]
```diff
@@ -4,28 +4,32 @@ const fs = require('fs');
...
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+app.use(cors());
 
-app.use(restify.bodyParser());
-app.use(restify.queryParser());
-
-fs.readdirSync('./middleware').map(name => {
-  require('./middleware/'+name)(app);
-});
```
Comment (davidbanham) [inline base/index.mustache]: How do middlewares get loaded if this function is removed?

File: base/test/routes/jwt_test.mustache [hunk 2]
```diff
@@ -24,14 +24,14 @@ describe('JWT authentication handling', () => {
   });
 
   it('must accept a valid JWT in a query string param', () => {
-    return fetch(url+'?jwt='+token, {
+    return fetch(url+'?token='+token, {
```
Comment (davidbanham) [inline base/test/routes/jwt_test.mustache]: Why is this required? Is it so that the express token parsing middleware picks it up correctly?

**Comment by davidbanham** [comment]
In general, looks great! Nice work. I have two questions that I've asked in line comments.

File: base/index.mustache (hunk 1 above)
Comment (nwinch) [inline base/index.mustache]: Yeah, I removed as this as after the cors and jwt middlewares were removed and now simply done with `app.use`, I figure if I we - as authors - are adding anymore it'd be the same way; either application level middleware, or routing level middleware.  Because there are no middleware generators, then there is no need to read and load them dynamically.

If a future redbeard user wants to add middleware, sure, they can add a separate folder for it if they so wish.

File: base/test/routes/jwt_test.mustache (hunk 2 above)
Comment (nwinch) [inline base/test/routes/jwt_test.mustache]: Yeah "jwt" would work fine too.  I changed it simply because I spent 20 mins brushing up on JWTs, and all terminology referenced "token" as the preferred generic label, and the fact that the variable name is "token", I decided to just change the query param :)

**Comment by nwinch** [comment]
@davidbanham Does my response on middleware make sense?  If yes, I'mma merge this sucker.

File: base/test/routes/jwt_test.mustache (hunk 2 above)
Comment (davidbanham) [inline base/test/routes/jwt_test.mustache]: Fair enough.

File: base/index.mustache (hunk 1 above)
Comment (davidbanham) [inline base/index.mustache]: I think it's worth having an opinion about where middleware should go. It saves people _having_ to re-make that decision every time. They're still welcome to make a different decision, because it's just code that they can edit. If we give a sane default, though, they're not forced to.

Also, while we are no longer adding middleware as part of base we're still likely to want redbeard middleware in other generators. User is the main area that springs to mind.

Comment (nwinch) [inline base/index.mustache]: Sure, if you think it's still useful, I'll add it back in.

Comment (davidbanham) [inline base/index.mustache]: :+1:

**Comment by davidbanham** [comment]
Will merge as soon as this passes CI.
